src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -1328,14 +1345,161 @@ private ItemStack forceItemStackToProvider(@NotNull final ICapabilityProvider pr
 
     //------------------------- Ending Required Tools/Item handling -------------------------//
 
+    //------------------------- Starting Assigned Citizen handling -------------------------//
+
     /**
-     * Sets the mirror of the current building.
+     * Get the main worker of the building (the first in the list).
+     *
+     * @return the matching CitizenData.
      */
-    public void invertMirror()
+    public CitizenData getMainCitizen()
     {
-        this.isBuildingMirrored = !isBuildingMirrored;
+        if (assignedCitizen.isEmpty())
+        {
+            return null;
+        }
+        return assignedCitizen.get(0);
+    }
+
+    /**
+     * Returns the worker of the current building.
+     *
+     * @return {@link CitizenData} of the current building
+     */
+    public List<CitizenData> getAssignedCitizen()
+    {
+        return new ArrayList<>(assignedCitizen);
     }
 
+    /**
+     * Method to remove a citizen.
+     *
+     * @param citizen Citizen to be removed.
+     */
+    public void removeCitizen(final CitizenData citizen)
+    {
+        if (isCitizenAssigned(citizen))
+        {
+            assignedCitizen.remove(citizen);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "remove()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)